### PR TITLE
Design tweaks to dialogs

### PR DESCRIPTION
**Pull request by dbkr**
Little bit of a mix of things in this one:
 * Support variable-width dialogs. Default is fixed-width as before,
   only UploadConformDialog is variable-width. Controlled by a prop
   to BaseDialog.
 * Fixes to the cancel 'x' - scale the mask image, tweak size & colour
 * Colour & boldness of dialog titles
 * Align the dialog title & cancel 'x'
 * Remove gap between dialog buttons & right hand side of dialog(!)
 * Round corners on dialogs
 * Add grey border on image preview in upload confirm dialog
 * and, squeezing in slightly randomly, finish the partially renamed
   ChatInviteDialog to AddressPickerDialog.

Screenshot 2019-04-03 at 16 23 37

### CODE REVIEW

**Comment by dbkr** [comment]
hold off on this for a bit - the room directory resizes itself in an annoying way

**Comment by dbkr** [comment]
ok, that's better

**Comment by turt2live** [comment]
screenshots? :D

**Comment by bwindels** [comment]
yeah, a screenshot would be nice to review this.

File: res/css/_common.scss [hunk 1]
```diff
@@ -381,6 +391,7 @@ textarea {
 .mx_Dialog button.mx_Dialog_primary, .mx_Dialog  input[type="submit"].mx_Dialog_primary {
     color: $accent-fg-color;
     background-color: $accent-color;
+    min-width: 156px;
```
Comment (jryans) [inline res/css/_common.scss]: Where does that size come from?

File: src/Modal.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ limitations under the License.
 const React = require('react');
 const ReactDOM = require('react-dom');
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
```
Comment (jryans) [inline src/Modal.js]: Unused import?

File: res/themes/dark/css/_dark.scss [hunk 3]
```diff
@@ -72,7 +72,7 @@ $avatar-bg-color: $bg-color;
 
 $h3-color: $primary-fg-color;
 
-$dialog-title-fg-color: #454545;
+$dialog-title-fg-color: #edf3ff;
```
Comment (jryans) [inline res/themes/dark/css/_dark.scss]: For the dark theme, I tried to make values flow from the top "unified palette" section if there was a match.

Maybe:

```suggestion
$dialog-title-fg-color: $base-text-color;
```

File: res/css/_common.scss (hunk 1 above)
Comment (dbkr) [reply res/css/_common.scss]: Just taken from the zeplin from what the size of the button was

File: src/Modal.js (hunk 2 above)
Comment (dbkr) [reply src/Modal.js]: ah, good spot. There were only a couple of linting errors in that file so I have just gone through & delinted it & removed it from the ignore list.

Comment (jryans) [reply src/Modal.js]: Wonderful, thanks for doing that! 😁

**Pull request merged**